pkg/resources/session_parameter.go

```diff
@@ -64,70 +68,59 @@ func CreateSessionParameter(d *schema.ResourceData, meta interface{}) error {
 		}
 	}
 
-	// add quotes to value if it is a string
-	typeString := reflect.TypeOf("")
-	if reflect.TypeOf(parameterDefault.DefaultValue) == typeString {
-		value = fmt.Sprintf("'%s'", snowflake.EscapeString(value))
-	}
-
-	builder := snowflake.NewSessionParameter(key, value, db)
-
 	onAccount := d.Get("on_account").(bool)
 	user := d.Get("user").(string)
+	parameter := sdk.SessionParameter(key)
 
+	var err error
 	if onAccount {
-		builder.SetOnAccount(onAccount)
+		err := client.Parameters.SetSessionParameterForAccount(ctx, parameter, value)
+		if err != nil {
+			return err
+		}
 	} else {
 		if user == "" {
 			return fmt.Errorf("user is required if on_account is false")
 		}
+		// add quotes to value if it is a string
+		typeString := reflect.TypeOf("")
+		if reflect.TypeOf(parameterDefault.DefaultValue) == typeString {
+			value = fmt.Sprintf("'%s'", snowflake.EscapeString(value))
+		}
+		builder := snowflake.NewSessionParameter(key, value, db)
 		builder.SetUser(user)
-	}
-
-	err := builder.SetParameter()
-	if err != nil {
-		return fmt.Errorf("error creating session parameter err = %w", err)
+		err = builder.SetParameter()
+		if err != nil {
+			return fmt.Errorf("error creating session parameter err = %w", err)
+		}
 	}
 
 	d.SetId(key)
 
-	var p *snowflake.Parameter
-	if onAccount {
-		p, err = snowflake.ShowAccountParameter(db, key)
-	} else {
-		p, err = snowflake.ShowSessionParameter(db, key, user)
-	}
-	if err != nil {
-		return fmt.Errorf("error reading session parameter err = %w", err)
-	}
-	err = d.Set("value", p.Value.String)
-	if err != nil {
-		return fmt.Errorf("error setting session parameter err = %w", err)
-	}
-	return nil
+	return ReadSessionParameter(d, meta)
 }
 
 // ReadSessionParameter implements schema.ReadFunc.
 func ReadSessionParameter(d *schema.ResourceData, meta interface{}) error {
 	db := meta.(*sql.DB)
-	key := d.Id()
+	client := sdk.NewClientFromDB(db)
+	ctx := context.Background()
+	parameter := d.Id()
 
 	onAccount := d.Get("on_account").(bool)
-	var p *snowflake.Parameter
 	var err error
+	var p *sdk.Parameter
 	if onAccount {
-		p, err = snowflake.ShowAccountParameter(db, key)
-		if err != nil {
-			return fmt.Errorf("error reading session parameter err = %w", err)
-		}
+		p, err = client.Sessions.ShowAccountParameter(ctx, sdk.AccountParameter(parameter))
```

Comment:
It's a bit weird to have `client.Sessions.ShowAccountParameter` and `client.Parameters.SetSessionParameterForAccount`, maybe we should have all parameter-related methods under the same object.

Reply:
in the snowflake api Account parameters are set at the  `ALTER ACCOUNT SET <>` while Session parameters are set using `ALTER SESSION SET <>` or for users with `ALTER USER SET <>`, or at the account level using the same `ALTER ACCOUNT SET <>`. This is a bit confusing. I think we should consolidate all the parameter methods into the client while still maintaining compatability with the original API. So `client.Parameters.SetSessionParameterForAccount()` would do the same thing as `client.Accounts.Alter(..)`. What we don't want it someone looking for the "ALTER SESSION SET PARAMETER" method and not being able to find it.

Reply:
All those function are part of `Parameters` interface now